<a href="https://colab.research.google.com/github/AhmedMansour1070/AhmedMansour1070/blob/main/Building_neural_network.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [31]:
from tensorflow.keras.datasets import mnist
(train_images , train_labels) , (test_images , test_labels) = mnist.load_data()

In [ ]:
train_images.shape
test_images.shape

In [ ]:
len(train_labels)
len(test_labels)

In [21]:
from tensorflow.keras import models
from tensorflow.keras import layers

In [24]:
network = models.Sequential()
network.add(layers.Dense(512, activation='relu', input_shape=(28 * 28,)))
network.add(layers.Dense(10, activation='softmax'))

In [32]:
network.compile(optimizer='rmsprop',
loss='categorical_crossentropy',
metrics=['accuracy'])

In [33]:
train_images = train_images.reshape((60000, 28 * 28))
train_images = train_images.astype('float32') / 255
test_images = test_images.reshape((10000, 28 * 28))
test_images = test_images.astype('float32') / 255

In [34]:
from tensorflow.keras.utils import to_categorical
train_labels = to_categorical(train_labels)
test_labels = to_categorical(test_labels)

In [35]:
network.fit(train_images, train_labels, epochs=5, batch_size=128)

Epoch 1/5
469/469 [==============================] - 4s 8ms/step - loss: 0.2557 - accuracy: 0.9258
Epoch 2/5
469/469 [==============================] - 4s 8ms/step - loss: 0.1021 - accuracy: 0.9698
Epoch 3/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0661 - accuracy: 0.9804
Epoch 4/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0481 - accuracy: 0.9857
Epoch 5/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0364 - accuracy: 0.9892


We quickly reach an accuracy of 0.989 (98.9%) on the training data. Now let’s
check that the model performs well on the test set

In [36]:
test_loss, test_acc = network.evaluate(test_images, test_labels)
print('test_acc:', test_acc)

313/313 [==============================] - 1s 2ms/step - loss: 0.0797 - accuracy: 0.9769
test_acc: 0.9768999814987183


That's overfitting!!